docbrown/core/src/graphview.rs

```diff
@@ -0,0 +1,306 @@
+use std::ops::{Range};
+use crate::graph::{TemporalGraph};
+use crate::graph::VertexView;
+use polars::prelude::*;
+use crate::Direction;
+use crate::tadjset::AdjEdge;
+
+type State = DataFrame;
+
+pub struct SingletonIterator<T: Copy>{value: T, exhausted: bool}
+
+impl<T: Copy> SingletonIterator<T> {
```

Comment:
why did you force T to be Copy?

Reply:
I am not actually using this any more, this was from messing around. I don't think the `Copy` is actually needed. It was an attempt at fixing a compile error though I think it did not actually make a difference.

Reply:
This isn't used I see, if you need to return a single item there is an iterator for that 
https://doc.rust-lang.org/std/iter/fn.once.html